src/Components/Components/src/ParameterAttribute.cs

```diff
@@ -11,5 +13,24 @@ namespace Microsoft.AspNetCore.Components
     [AttributeUsage(AttributeTargets.Property, AllowMultiple = false, Inherited = false)]
     public sealed class ParameterAttribute : Attribute
     {
+        /// <summary>
+        /// Gets or sets a value that determines whether the parameter will capture the values of
+        /// additional attributes that don't match any other parameter.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// <see cref="CaptureExtraAttributes"/> allows a component to accept arbitrary additional
+        /// attributes, and pass them to another component, or some element of the underlying markup.
+        /// </para>
+        /// <para>
+        /// <see cref="CaptureExtraAttributes"/> can be used on at most one parameter per component.
+        /// </para>
+        /// <para>
+        /// <see cref="CaptureExtraAttributes"/> should only be applied to parameters of a type that
+        /// can be used with <see cref="RenderTreeBuilder.AddMultipleAttributes{T}(int, System.Collections.Generic.IEnumerable{System.Collections.Generic.KeyValuePair{string, T}})"/>
+        /// such as <see cref="Dictionary{String, Object}"/>.
+        /// </para>
+        /// </remarks>
+        public bool CaptureExtraAttributes { get; set; }
```

Comment:
Suggestions for name?

I feel like making this an option on `[Parameter]` is the best choice. We want this to be settable as a normal parameter as well.

Reply:
I'd prefer we call these *parameters*, not attributes, since the use cases for this feature is more general than just splatting attributes. For example you might want to capture all extra supplied parameters programmatically then run some procedural logic over that dictionary, and not necessarily splat them back out at all.

So, I'd be open to any of:

 1. `[Parameter(CaptureExtraParameters = true)]`
 2. `[Parameter(CaptureUnboundParameters = true)]`
 3. `[Parameter(CaptureUnmatchedParameters = true)]`
 4. `[UnboundParameters]` (where `UnboundParameters` inherits from `[Parameter]`, so you can still pass it explicitly)
 5. `[UnmatchedParameters]` (where `UnmatchedParameters` inherits from `[Parameter]`, so you can still pass it explicitly)

I think I prefer them in reverse order of that list (i.e., `[UnmatchedParameters]` as my favourite). If you think it's problematic to inherit from `[Parameter]` then a bool flag is fine with me too.

Most .NET devs would think of "unbound" as more intuitive than "unmatched", but it's always annoying to overload termonology hence my reluctance to reuse the notion of "binding".

Reply:
I expect it's clear, but just in case, I think it's legitimate to call these "unmatched parameters" as opposed to "unmatched arguments" based on the "actual parameters vs formal parameters" distinction rather than the "arguments vs parameters" distinction.

Reply:
One last option, in case the "parameters" vs "arguments" thing is too problematic:

```
[Parameter(CaptureUnmatchedValues = true)]
```

Reply:
> I expect it's clear, but just in case, I think it's legitimate to call these "unmatched parameters" as opposed to "unmatched arguments" based on the "actual parameters vs formal parameters" distinction rather than the "arguments vs parameters" distinction.

I don't think you and I will ever agree 100% on the debate between parameters and arguments - and I'm not trying to start that discussion again 😆 I still really prefer terminology about `Values` vs `Parameters` and maybe that's a point of common ground.

--

The things that I don't like about `[UnmatchedParameters]` is that inheritance is problematic with the way we use these attributes (and there are examples of this in TagHelpers). 

We expect to add more API surface to these attributes that we will statically analyze with the compiler/analyzers and other tools. To make that work it's really important that we don't let the user **run code** to affect the settings we care about, it has to be purely declarative. The constructor of a defined class (if we unseal the attribute) is the only place a user can do this. 

People did this with TagHelpers, and found that it worked in the runtime, as long as you use runtime compilation. It did not work in the compiler or at build time (which we added later).

So I guess we could tell people that's not supported, but there's other wierdness with unsealed *and* non-abstract attribute types that I'd like to avoid.

----

Basically I think the best option of these is:

```C#
[Parameter(CaptureUnmatchedValues = true)]
```

Unless we find a much better idea than this, I'm going to go with it for now. We can debate this more once we can start writing code with it.